src/psyclone/psyGen.py

```diff
@@ -5575,9 +5578,14 @@ def _case_construct_handler(self, node, parent):
                 "Failed to find closing case statement in: "
                 "{0}".format(str(node)))
 
-        # Search for all the CASE clauses in the Case_Construct
+        # Search for all the CASE clauses in the Case_Construct. We do this
+        # because the fp2 parse tree has a flat structure at this point with
+        # the clauses being siblings of the contents of the clauses. The
+        # final index in this list will hold the position of the end-select
+        # statement.
         clause_indices = []
         selector = None
+        default_clause_idx = -1
         for idx, child in enumerate(node.content):
```

Comment:
Update `# Case Default and value Ranges not supported yet, if found` comment

Reply:
Oh yes, well spotted.